src/compiler/interface.jl

```diff
@@ -123,26 +144,15 @@ julia> haskey(g, z)  # only x and y are parameters
 false
 ```
 """
-gradient
-
-"""
-    Params([A, B])
-
-Container for implicit parameters, used when differentiating
-a zero-argument funtion `() -> loss(A, B)` with respect to `A, B`.
-"""
-struct Params
-  order::Buffer # {Any, Vector{Any}}
-  params::IdSet{Any} # TODO store ids only
+function gradient(f, ps::Params)
+  y, back = pullback(f, ps)
+  back(sensitivity(y))
```

Comment:
`struct Params` hasn't changed. I just had to make a separate method `gradient(f, ps::Params)` and move that below it, the diff is confused. 

The reason that's separate is that projection doesn't happen for Params at all right now. But perhaps it should.